tracer/src/Datadog.Trace.Tools.Runner/GacUninstallCommand.cs

```diff
@@ -39,34 +39,48 @@ private void Execute(InvocationContext context)
             return;
         }
 
-        using var container = NativeMethods.CreateAssemblyCache();
-        var hr = container.AssemblyCache.UninstallAssembly(UninstallAssemblyFlags.None, assemblyName!, IntPtr.Zero, out var position);
+        using var gacMethods = GacNativeMethods.Create();
+        var assemblyCache = gacMethods.CreateAssemblyCache();
 
-        switch (position)
+        var result = Hresult.S_OK;
+        var installedAssemblyNames = gacMethods.GetAssemblyNames(assemblyName!);
+        if (installedAssemblyNames.Count == 0)
         {
-            case UninstallDisposition.IASSEMBLYCACHE_UNINSTALL_DISPOSITION_ALREADY_UNINSTALLED:
-                Utils.WriteWarning($"Assembly '{assemblyName}' was already uninstalled from the GAC.");
-                context.ExitCode = hr;
-                break;
-            case UninstallDisposition.IASSEMBLYCACHE_UNINSTALL_DISPOSITION_REFERENCE_NOT_FOUND:
-                Utils.WriteWarning($"Assembly '{assemblyName}' not found in the GAC.");
-                context.ExitCode = hr;
-                break;
-            case UninstallDisposition.IASSEMBLYCACHE_UNINSTALL_DISPOSITION_STILL_IN_USE:
-                Utils.WriteWarning($"Assembly '{assemblyName}' is still in use.");
-                context.ExitCode = hr;
-                break;
+            Utils.WriteWarning($"Assembly '{assemblyName}' is not installed the GAC.");
```

Comment:
Should this just count as success? Afterall, the file isn't installed, which is what you really want, right? i.e. Idempotency?

Reply:
True, I've made the change here: [Add exit code](https://github.com/DataDog/dd-trace-dotnet/pull/6995/commits/6bcc865c48c67b8915b0c2a1b23c3dbc86e13aae)

Reply:
Wait, but you made it an _error_ code? Shouldn't this be success? 😅